protocol_tests/connection/test_connection.py

```diff
@@ -4,13 +4,314 @@
+        # If we're handling a DID exchange
+        if preferred_handshake_protocol in OutOfBandInvite.DID_EXCHANGE_TYPES:
+
+            # Make a DID exchange request
+            request = DidExchangeRequest.make(
+                conn.did,
+                conn.verkey_b58,
+                config['endpoint'],
+                conn.sigkey,
+                'test-oob-connection-started-by-suite-did-exchange'
+            )
+
+            yield 'did_exchange_request', request
+            # Send out the DID exchange request, wait for the response and validate it
+            response = DidExchangeResponse(await conn.send_and_await_reply_async(
+                request,
+                condition=lambda msg: msg.type == (DidExchangeResponse.TYPE or DidExchangeResponse.ALT_TYPE),
+                timeout=10
+            ))
+            response.validate()
+
+            yield 'did_exchange_response', response
+
+            # We need to update the connection information here so we can
+            # properly contact the agent on the other side
+            new_info = response.get_connection_info()
+            conn.update(**new_info._asdict())
+
+            yield 'new_connection_info', new_info, conn
+
+            # Next, to finish out the flow we have to send out the did exchange compete msg
+            complete = {
+                "@type": "https://didcomm.org/didexchange/1.0/complete",
+                "~thread": {
+                    "thid": request["@id"],
+                    "pthid": invite["@id"],
+                    "sender_order": 1
+                }
+            }
+            await conn.send_async(complete)
+
+            yield 'did_exchange_complete', complete, conn
+
+        # Otherwise, we're dealing with a connection protocol
+        elif preferred_handshake_protocol in OutOfBandInvite.CONNECTION_TYPES:
+
+            # Same flow here, except with a different protocol.
+            # Make a request, wait for the response, validate the response and update the connection
+            request = ConnectionRequest.make(
+                'test-oob-connection-started-by-suite-connections',
+                conn.did,
+                conn.verkey_b58,
+                config['endpoint']
+            )
+
+            yield 'request', request
+
+            response = ConnectionResponse(await conn.send_and_await_reply_async(
+                request,
+                condition=lambda msg: msg.type == (ConnectionResponse.TYPE or ConnectionResponse.ALT_TYPE),
+                timeout=30
+            ))
+
+            yield 'response', response
+
+            response.validate_pre_sig_verify()
+            response.verify_sig(info.recipients[0])
+            response.validate_post_sig_verify()
+
+            new_info = response.get_connection_info()
+            conn.update(**new_info._asdict())
+
+
+        # Finally, we'll finish up with a trust ping to validate that the connection works
+        await conn.send_and_await_reply_async(
```

Comment:
It is possible that someone supports the OOB protocol but doesn't support trust ping, so I don't think this should be here.  I think the complete message will validate that the connection works.